flatcar-postinst

```diff
@@ -30,303 +26,185 @@ case "${INSTALL_LABEL}" in
-	[8a7b03d92a8b115943e7f004820fadd2dc6ab125c077a48fb232a1e9ac77fdb27fbb01d52fd33a6ddf65a9f58ce981244c99bcca821030511caa277bc2f68239]='\x74\x02\x00\x00'
-	[a3e9dadfe3cc34189b5fee83bfc01c3c5b42e04ba19cdcd84f8301c42566617b5916294e2414348139d8c5e557a7ccf6c0d3dca0661f2d10c0c0077345630b1d]='\x74\x02\x00\x00'
-	[c127d7c1dbd5d11cf7af627e37808ea16166b6430ddd8e96111e503cc78ae1fd78083d474495951743fa1b489140be63178a4bb65dabb0d719c5d0ad9c57eb78]='\x74\x02\x00\x00'
-	[f1f9abefa49eeba6a3fe46ba3d254dfc3fa6e2cd8823835e2d982e9cbcd0d82c298e3896dc79d234305d2262a828a7398526906022f8ed7407368725d95e08d8]='\x74\x02\x00\x00'
-	)
-
-	filesum="$(sha512sum "${tmpfile}" | cut -d' ' -f1)"
-	if [[ -z "${offsets[$filesum]}" ]]; then
-		echo "Nothing to do"
-		rm "${tmpfile}"
-		touch "${escape_hatch}"
-		return
-	fi
-
-	printf "${correctvals[$filesum]}" | dd of="${tmpfile}" bs=1 seek="${offsets[$filesum]}" conv=notrunc status=none
-
-	# There's a lot of sync'ing going on. On remotely up to date systems (newer than 1109.1.0), sync can operate on
-	# individual files. On old systems it syncs everything which is slow, but we want to be as safe as possible.
-	# Since we write out the escape hatch after everything is done, this will only happen once.
-
-	# rezip onto /boot so ENOSPC can't cause problems
-	gzip -c "${tmpfile}" > "${file}.tmp"
-	rm "${tmpfile}"
-
-	# in case something goes horribly wrong. Do not use mv -b since it moves the original file to
-	# the backup name then moves the new file to the target, leaving a window with no file
-	cp -p "${file}.mod" "${file}.bak.bug2400"
-	sync "${file}.bak.bug2400" "${file}.tmp"
-
-	# use mv then sync to be as atomic as possible
-	mv "${file}.tmp" "${file}.mod"
-	sync '/boot/coreos/grub/i386-pc/'
-
-	touch "${escape_hatch}"
-
-	echo 'linux.mod updated successfully'
+OEMID=$({ grep -m 1 -o "^ID=.*" "${OEM_MNT}"/oem-release || true ; } | cut -d = -f 2)
+
+# Declare which OEM sysext images are expected to exist, and the files to delete in the initramfs migration action (Use /oem instead of /usr/share/oem/ to avoid symlinks from the initrd.)
+# Note: Because associative arrays can't hold lists, we store the files newline-separated (not simply space separated, it must be only one file per line!)
+NVIDIA_FILES="/etc/systemd/system/nvidia.service
+/oem/bin/setup-nvidia
+/oem/bin/install-nvidia
+/oem/units/nvidia.service"
+declare -A OEM_SYSEXTS
+OEM_SYSEXTS[qemu]=""
+OEM_SYSEXTS[azure]="${NVIDIA_FILES}
+/etc/systemd/system/oem-cloudinit.service
+/etc/systemd/system/multi-user.target.wants/oem-cloudinit.service
+/etc/systemd/system/waagent.service
+/etc/systemd/system/multi-user.target.wants/waagent.service
+/oem/waagent.conf
+/oem/python/
+/oem/bin/
+/oem/units/
+/oem/base/
+"
+# TODO: For new OEM sysext images, declare an entry like OEM_SYSEXTS[ami]="${NVIDIA_FILES}..." or even just OEM_SYSEXTS[qemu]="" to enable fallback downloads for old clients
+# TODO: Keep in sync with bin/flatcar-update from the "init" repo
```

Comment:
I'm starting to think, that these lists should be installed somewhere in `/usr/share/flatcar/oem-stuff` directory. That way, the changes for the specific OEM sysext could be limited to just the `scripts` repo (or even, to the `coreos-overlay` part of it). Possibly flatcar-update could use it too.

So could having something like `/usr/share/flatcar/oem-stuff/oems/<OEMID>/obsolete-files` be suitable for our needs? I suppose it could be installed by `coreos-base/oem-obsolete-files` package - it would be built by the packages job, so it would be a part of a generic image.

Reply:
Good idea, maybe instead of starting one more new package we can use misc-files or coreos-init.